commit/merkleroot/rmn/client.go

```diff
@@ -19,13 +33,533 @@ type Client interface {
 	ComputeReportSignatures(
 		ctx context.Context,
 		destChain *rmnpb.LaneDest,
-		requestedUpdates []rmnpb.FixedDestLaneUpdateRequest,
+		requestedUpdates []*rmnpb.FixedDestLaneUpdateRequest,
 	) (*ReportSignatures, error)
 }
 
 type ReportSignatures struct {
 	// ReportSignatures are the ECDSA signatures for the lane updates for each node.
+	Signatures []*rmnpb.EcdsaSignature
+	// LaneUpdates are the lane updates for each source chain that we got the signatures for.
 	// NOTE: A signature[i] corresponds to the whole LaneUpdates slice and NOT LaneUpdates[i].
-	Signatures  []rmnpb.EcdsaSignature
-	LaneUpdates []rmnpb.FixedDestLaneUpdate
+	LaneUpdates []*rmnpb.FixedDestLaneUpdate
+}
+
+// PBClient is the base RMN Client implementation.
+type PBClient struct {
+	lggr                                    logger.Logger
+	rawRmnClient                            RawRmnClient
+	rmnNodes                                []RMNNodeInfo
+	rmnRemoteAddress                        []byte
+	rmnHomeConfigDigest                     []byte
+	minObservers                            int
+	minSigners                              int
+	observationsInitialRequestTimerDuration time.Duration
+	reportsInitialRequestTimerDuration      time.Duration
+}
+
+// RMNNodeInfo contains the information about an RMN node.
+type RMNNodeInfo struct {
+	ID                    uint32 // ID is the index of this node in the RMN config
```

Comment:
nit:
```suggestion
    // ID is the index of this node in the RMN config
	ID                    uint32 
```

Reply:
not sure i follow, what's the suggestion, dmed